packages/components/src/date-time/date/index.tsx

```diff
@@ -100,139 +60,284 @@ function DatePickerDay( { day, events = [] }: DatePickerDayProps ) {
 export function DatePicker( {
 	currentDate,
 	onChange,
-	events,
+	events = [],
 	isInvalidDate,
 	onMonthPreviewed,
 	startOfWeek = 0,
 }: DatePickerProps ) {
-	const nodeRef = useRef< HTMLDivElement >( null );
+	const date = currentDate ? inputToDate( currentDate ) : new Date();
 
-	const onMonthPreviewedHandler = ( newMonthDate: Moment ) => {
-		onMonthPreviewed?.( newMonthDate.toISOString() );
-		keepFocusInside();
-	};
+	const {
+		calendar,
+		viewing,
+		setSelected,
+		setViewing,
+		isSelected,
+		viewPreviousMonth,
+		viewNextMonth,
+	} = useLilius( {
+		selected: [ startOfDay( date ) ],
+		viewing: startOfDay( date ),
+		weekStartsOn: startOfWeek,
+	} );
 
-	/*
-	 * Todo: We should remove this function ASAP.
-	 * It is kept because focus is lost when we click on the previous and next month buttons.
-	 * This focus loss closes the date picker popover.
-	 * Ideally we should add an upstream commit on react-dates to fix this issue.
-	 */
-	const keepFocusInside = () => {
-		if ( ! nodeRef.current ) {
-			return;
-		}
-
-		const { ownerDocument } = nodeRef.current;
-		const { activeElement } = ownerDocument;
+	// Used to implement a roving tab index. Tracks the day that receives focus
+	// when the user tabs into the calendar.
+	const [ focusable, setFocusable ] = useState( startOfDay( date ) );
 
-		// If focus was lost.
-		if (
-			! activeElement ||
-			! nodeRef.current.contains( ownerDocument.activeElement )
-		) {
-			// Retrieve the focus region div.
-			const focusRegion = nodeRef.current.querySelector(
-				'.DayPicker_focusRegion'
-			);
-			if ( ! ( focusRegion instanceof HTMLElement ) ) {
-				return;
-			}
-			// Keep the focus on focus region.
-			focusRegion.focus();
-		}
-	};
+	// Allows us to only programmatically focus() a day when focus was already
+	// within the calendar. This stops us stealing focus from e.g. a TimePicker
+	// input.
+	const [ isFocusWithinCalendar, setIsFocusWithinCalendar ] =
+		useState( false );
 
-	const onChangeMoment = ( newDate: Moment | null ) => {
-		if ( ! newDate ) {
-			return;
-		}
+	// Update internal state when currentDate prop changes.
+	const [ prevCurrentDate, setPrevCurrentDate ] = useState( currentDate );
+	if ( currentDate !== prevCurrentDate ) {
+		setPrevCurrentDate( currentDate );
+		setSelected( [ startOfDay( date ) ] );
+		setViewing( startOfDay( date ) );
+		setFocusable( startOfDay( date ) );
+	}
 
-		// If currentDate is null, use now as momentTime to designate hours, minutes, seconds.
-		const momentDate = currentDate ? moment( currentDate ) : moment();
-		const momentTime = {
-			hours: momentDate.hours(),
-			minutes: momentDate.minutes(),
-			seconds: 0,
-		};
+	return (
+		<div
+			className="components-datetime__date"
+			role="application"
+			aria-label={ __( 'Calendar' ) }
```

Comment:
I copied this from the previous implementation but I'm not sure if it's the right thing to do. Maybe someone who knows what they're doing can answer cc. @talldan @tellthemachines